src/interface/container_collection.cpp

```diff
@@ -17,6 +17,33 @@
 
 namespace parthenon {
 
+template <typename T>
```

Comment:
This isn't your job  and it shouldn't go into this PR. But we have a pattern where we repeat the same code like this for each type of variable/swarm/etc. I wonder if there's a way to make it generic.

Reply:
I know what you mean but I don't see immediately how to make this generic. Anyway, thank you for allowing me to ignore this on this PR :)